lib/model/services/ble/machine_service.dart

```diff
@@ -594,11 +618,15 @@ class EspressoMachineService extends ChangeNotifier {
       currentShot.targetEspressoWeight = settingsService.targetEspressoWeight;
 
       _delayedStopActive = false;
+      _weightMoveOnFrames = {};
       isPouring = false;
       shotList.clear();
+      weightMeasurementsDuringShot.clear();
       baseTime = DateTime.now().millisecondsSinceEpoch / 1000.0;
       log.info("basetime $baseTime");
       lastPourTime = 0;
+
+      Timer.periodic(const Duration(milliseconds: 50), checkMoveOnAtWeight);
     }
```

Comment:
50ms inteval seems very often... Is that really needed? When will the timer stopped?

Reply:
Without cancel of the timer, it will be a BIG memory and CPU leak... This function is called many times.

Reply:
It’s cancelled at the end of the shot

Reply:
Ah I see that you implicitely send the timer to the callback function. Wish it would be more readable via
timer => callback(timer) even if it is not needed. Makes it more clear to the reader ^^

Reply:
I can do that!